airbyte-integrations/connectors/destination-clickhouse-v2/src/main/kotlin/io/airbyte/integrations/destination/clickhouse_v2/client/ClickhouseAirbyteClient.kt

```diff
@@ -0,0 +1,94 @@
+package io.airbyte.integrations.destination.clickhouse_v2.client
+
+import com.clickhouse.client.api.Client
+import com.clickhouse.client.api.data_formats.ClickHouseBinaryFormatReader
+import com.clickhouse.data.ClickHouseDataType
+import io.airbyte.cdk.load.client.AirbyteClient
+import io.airbyte.cdk.load.data.AirbyteType
+import io.airbyte.cdk.load.data.ArrayType
+import io.airbyte.cdk.load.data.ArrayTypeWithoutSchema
+import io.airbyte.cdk.load.data.BooleanType
+import io.airbyte.cdk.load.data.DateType
+import io.airbyte.cdk.load.data.IntegerType
+import io.airbyte.cdk.load.data.NumberType
+import io.airbyte.cdk.load.data.ObjectType
+import io.airbyte.cdk.load.data.ObjectTypeWithEmptySchema
+import io.airbyte.cdk.load.data.ObjectTypeWithoutSchema
+import io.airbyte.cdk.load.data.StringType
+import io.airbyte.cdk.load.data.TimeTypeWithTimezone
+import io.airbyte.cdk.load.data.TimeTypeWithoutTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithoutTimezone
+import io.airbyte.cdk.load.data.UnionType
+import io.airbyte.cdk.load.data.UnknownType
+import io.airbyte.cdk.load.orchestration.db.TableName
+import io.airbyte.integrations.destination.clickhouse_v2.spec.ClickhouseConfiguration
+import io.airbyte.integrations.destination.clickhouse_v2.write.direct.ClickhouseDirectLoadSqlGenerator
+import io.github.oshai.kotlinlogging.KotlinLogging
+import jakarta.inject.Singleton
+
+val log = KotlinLogging.logger {  }
+
+@Singleton
+class ClickhouseAirbyteClient(private val client: Client
+    ): AirbyteClient<ClickHouseDataType>() {
+    override fun getNumberOfRecordsInTable(tableName: TableName): Long? {
```

Comment:
I don't think we actually need number of records, but just whether there is at least one record. So the name should reflect that.

Reply:
This method is actually returning the number of records. Then `getTableStatus` is using it to check if the table exists and if it has at least 1 record in it.